app/src/main/java/org/wikipedia/ActivityLifecycleHandler.java

```diff
@@ -23,6 +27,22 @@ public void onActivityCreated(Activity activity, Bundle savedInstanceState) {
         if (activity instanceof MainActivity) {
             haveMainActivity = true;
         }
+        if (Prefs.shouldMatchSystemTheme() && Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
```

Comment:
If we choose to run this logic onConfigurationChanged() of BaseActivity, every child activity will have to include android:configChanges=uimode in its manifest declaration, it is not work when added to BaseActivity. this might be error prone if we forget to add it while creating new activities. So I feel it is better to take advantage of the existing lifecycle handler.